### DISCUSSION TITLE
Build Surelog on Windows and encounter C:\Users\Brandon Huang\Desktop\Surelog\build\third_party\UHDM\GenerateCode.vcxproj 9009 error

### DISCUSSION THREAD

**Discussion by brandot4life** [chipsalliance/UHDM#1092] [Q&A]
Dear all,
 I'm trying to install, build and use Surelog on windows, but when I try to build it by cmake 
cmake .. -G "Visual Studio 17 2022" -A x64
cmake --build . --config Release
 it comes out that 
 C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: 'C:\Users\Brandon Huang\Desktop\Surelog\build\CMakeFile
s\09664403dd9f0afc4c40ecaf8d0fc72c\UHDM.capnp.h.rule;C:\Users\Brandon Huang\Desktop\Surelog\build\CMakeFiles\82992fbb4833a276fe96b6dab66bdf84\GenerateCode.rule' 的自訂建置已結束，代碼為 9009。 [C:\User
s\Brandon Huang\Desktop\Surelog\build\third_party\UHDM\GenerateCode.vcxproj]

And I don't know why? Can you please give me some advice for that?
Thanks in advanced!!!

**Comment by hs-apotell** [comment]
Visual Studio project generation is not a supported workflow. We recommend using CMake support for VS. Just open the folder in VS and the dev environment will automatically configure your project, ready to build.

**Comment by brandot4life** [comment]
Hi all,
  Thanks to everyone's help, I have successfully built Surelog on the
virtual machine and used UHDM to display the parsing result.

However, during the analysis of the architecture, I found that information
like *port ranges* doesn't seem to connect directly with the *port names*.
As a result, it seems unclear which range numbers belong to which port
names.

From my current understanding, it is possible to find the port name and its
connections to other modules through vpiPort. However, the port's *range
numbers* seem to be displayed separately in vpiRange, and it might not be
clear which port they belong to.

May I ask if there is a way to establish the relationship between these two
so that they can be connected?

I sincerely appreciate you taking the time to reply. Thank you very much!

BRs,

Brandon Huang

HS ***@***.***> 於 2024年11月1日 週五 上午5:57寫道：
